### DISCUSSION TITLE
Hi ! sorry for the help I ask for, it may not be your first time.

### DISCUSSION THREAD

**Discussion by Alp136** [bangcorrupt/hacktribe#272] [Q&A] [answered]
Hi ! sorry for the help I ask for, it may not be your first time.

So english is not my first language, and I just absolutely don't understand what i'm doing.
I've tried to install the hacktribe editor, and the hacktribe-gui.exe, when I follow the instructions, it says "File not found: hacktribe\hash\SYSTEM.VSB.sha
Source hash file not found"

I've already downloaded the vsb from korg website, the electribe sampler 2.02 one, I don't know what I'm doing wrong.

I already tried to follow the special instruction you gave in the Q&A, but I don't understand anything, I also tried to install the dependancies, but I'm lost with all the files to download, and I don't know what to do with them..

Thanks for the help !

Also, isn't there a way to simply dowload the already hacked version of the vsb ? I don' t mean to insult anyone's work, I just don't understand why it is so hard

_Originally posted by @Alp136 in https://github.com/bangcorrupt/hacktribe/discussions/270_

**Comment by AndrewSosna** [comment]
See @bangcorrupt 's answer here for the reason: https://github.com/bangcorrupt/hacktribe/discussions/70#discussioncomment-6471355
There probably is a way to download hacked file somewhere else, but not here.

**Comment by bangcorrupt** [comment, marked as answer]
> "File not found: hacktribe\hash\SYSTEM.VSB.sha
> Source hash file not found"

If you clone the `hacktribe-editor` repo recursively it should get all the files in the right place:

    git clone --recursive https://github.com/bangcorrupt/hacktribe-editor.git

If you downloaded the repo as a zip archive from the Github website, you will also need to download the main `hacktribe` repo and put its contents at `hacktribe-editor/hacktribe`.

If you run `hacktribe-gui.exe` you shouldn't need to install any dependencies, just put the factory firmware at `hacktribe-editor/hacktribe/SYSTEM.VSB`.  If you are running the GUI from the command line, you will need to be in the `hacktribe-editor` directory for it to find the files.

Alternatively, I started work on a [tutorial](https://github.com/bangcorrupt/hacktribe-tutorial).  It's probably more reliable than the GUI.

**Reply by AndrewSosna** [reply]
I don't know, based on my experience as a developer, users don't read tutorials, or read them fast, skipping important notes :) Especially because we're talking about python with dependencies, environments, etc, much more potential sources of errors.
I feel like the best way is to make some 100% safe GUI/Wizard/whatever so it's impossible to make any error. I was thinking about making a native patcher for macOS, wizard style, where user would answer step by step "do you have Sampler or Synth or you don't know? ok, is it red or blue or gray or black?" and so on,  but I didn't have time yet because of my work and I didn't suggest this for you because patcher is a secondary tool relative to the patch itself

On the other hand, more people can use it, more people can be interested, involved, support you etc.

**Reply by Alp136** [reply]
I just read the turorial ! and clearly, I feel more confident to try again, thanks a lot for HackTribe !
I did find a hacked firmware online and it worked, but it's an older version, so I'll try again but following your tutorial, big thanks !

**Reply by bangcorrupt** [reply]
> users don't read tutorials

Yes.  I wrote the tutorial as a bash script, so it can be blindly run in a codespace and produce the correct result.  There is still the barrier of learning to run a script, but it's a useful skill.

> I was thinking about making a native patcher 

This would be really helpful, I'm terrible at GUI programming. I started work adding patching support to `ht-cli`.  I think it would be best if it set up the top level `KORG` directory for firmware swapping with all the samples and patterns edited and in the right place.  Then we just copy the directory to the SD card to update.

> didn't have time yet

I feel that.


> I feel more confident to try again

Let me know if any of it doesn't make sense.  I tried to make it so it's clear from the screenshots what you need to do.
